src/ts/address-search.ts

```diff
@@ -1,1240 +1,1441 @@
+      this.currentDataSet = countryCodeAndDataset[1];
+    }
+
+    // (Re-)set the property stating whether the search input has been reset.
+    // This is needed for instances when the search input is also an address
+    // output field. After an address has been returned, you don't want a new 
+    // search being triggered until the field has been cleared.
+    if (this.currentSearchTerm === '') {
+      this.hasSearchInputBeenReset = true;
+    }
+
+    // Check if searching is permitted
+    if (this.canSearch()) {
+      // Abort any outstanding requests
+      if (this.request.currentRequest) {
+        this.request.currentRequest.abort();
+      }
+
+      var regex = /^\/{0,}(?:[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+|[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3})$/;
+
+      if (regex.test(this.currentSearchTerm) && this.options.enableWhat3Words && this.what3WordCountries.indexOf(this.currentCountryCode) > -1) {
+        this.isWhat3Words = true;
+      }
+      else {
+        this.isWhat3Words = false;
+      }
+
+      // Fire an event before a search takes place
+      this.events.trigger('pre-search', this.currentSearchTerm);
+
+      // Construct the new Search URL and data
+      const data = this.isWhat3Words ? this.generateLookupDataForApiCall(this.currentSearchTerm, true) : this.generateSearchDataForApiCall();
+
+      // Store the last search term
+      this.lastSearchTerm = this.currentSearchTerm;
+
+      // Hide the inline search spinner
+      this.searchSpinner.hide();
+
+      // Show an inline spinner whilst searching
+      this.searchSpinner.show();
+
+      // Set the API URL, headers and callback function depending on the search type
+      if (this.isWhat3Words) {
+        const url = this.baseUrl + this.lookupEndpoint;
+        const headers = [];
+        const callback = this.picklist.showWhat3Words;
+
+        // Initiate new Search request
+        this.request.send(url, 'POST', callback, data, headers);
+      } else {
+        const url = this.baseUrl + (this.searchType === AddressValidationMode.VALIDATE ? this.validateEndpoint : this.searchEndpoint);
+        const headers = this.searchType === AddressValidationMode.VALIDATE ? [{ key: 'Add-Metadata', value: true }] : [];
+        const callback = this.searchType === AddressValidationMode.VALIDATE ? this.result.handleValidateResponse : this.picklist.show;
+
+        // Initiate new Search request
+        this.request.send(url, 'POST', callback, data, headers);

```

Comment:
This line is the same in both branches so could've come outside the `if/else`

Reply:
fixed